lab-sarah/server.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+//node modules
+const url = require('url');
+const http = require('http');
+const queryString = require('querystring');
+
+//npm modules
+const cowsay = require('cowsay');
+
+//module constants
+const PORT = process.env.PORT || 3000;
+
+const parseBody = require('./lib/parse-body.js');
+
+//module logic
+const server = http.createServer(function(req, res) {
+  // console.log('BEGINNING', res, 'END OF RESPONSE');
+  req.url = url.parse(req.url);
+  req.url.query = queryString.parse(req.url.query);
+
+  if(req.url.pathname ==='/') {
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    res.write('hello world');
+    res.end();
+    return;
+  }
+
+  if(req.method === 'GET' && req.url.pathname === '/cowsay') {
+
+    if (req.url.query.text) {
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      //respond with a body including the value returned from the cowsay.say
+      //basically what the person typed into the query
+      var cowText = req.url.query.text;
```

Comment:
You could also pass in req.url.query directly into the cowsay.say method, as it's equivalent to an object with a property of text and a value of a string, which is what that method expects.